File: yasmin/tests/python/test_cb_state.py
# Copyright (C) 2023 Miguel Ángel González Santamarta

# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.

# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.


import unittest
from yasmin import CbState


class TestCbState(unittest.TestCase):

    def setUp(self):

        def execute(blackboard):
            return "outcome1"

        self.state = CbState(["outcome1"], execute)

    def test_cb_state_call(self):
        self.assertEqual("outcome1", self.state())

File: yasmin/test/test_cb_state.cpp
// Copyright (C) 2025 Miguel Ángel González Santamarta
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#include <gtest/gtest.h>

#include <memory>
#include <string>

#include "yasmin/blackboard.hpp"
#include "yasmin/cb_state.hpp"

using namespace yasmin;

class TestCbState : public ::testing::Test {
protected:
  CbState::SharedPtr state;
  yasmin::Blackboard::SharedPtr blackboard;

  void SetUp() override {
    blackboard = yasmin::Blackboard::make_shared();

    auto execute = [](yasmin::Blackboard::SharedPtr bb) {
      return std::string("outcome1");
    };

    state = CbState::make_shared(std::set<std::string>{"outcome1"}, execute);
  }
};

TEST_F(TestCbState, TestCbStateCall) {
  EXPECT_EQ((*state)(blackboard), "outcome1");
}

int main(int argc, char **argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}
